cmake/FindVulkanHeaders.cmake

```diff
@@ -89,36 +89,42 @@ else()
-        string(REGEX REPLACE "[^\)]+$" "" VulkanHeaders_version_2 ${VulkanHeaders_version_1})
-        # Erase the parentheses themselves, and strip any spaces
-        string(REPLACE "(" "" VulkanHeaders_version_3 ${VulkanHeaders_version_2})
-        string(REPLACE ")" "" VulkanHeaders_version_4 ${VulkanHeaders_version_3})
-        string(REPLACE " " "" VulkanHeaders_version_5 ${VulkanHeaders_version_4})
-        # Split up the versions into major/minor
-        string(REGEX REPLACE "^([0-9]+)\\,[0-9]+\\,[0-9]+" "\\1" VulkanHeaders_cur_major "${VulkanHeaders_version_5}")
-        string(REGEX REPLACE "^[0-9]+\\,([0-9]+)\\,[0-9]+" "\\1" VulkanHeaders_cur_minor "${VulkanHeaders_version_5}")
-        # Compare to only update to a newer version
+foreach(VulkanHeaders_line ${VulkanHeaders_lines})
+
+    # First, handle the case where we have a major/minor version
+    #   Format is:
+    #        #define VK_API_VERSION_X_Y VK_MAKE_VERSION(X, Y, 0)
+    #   We grab the major version (X) and minor version (Y) out of the parentheses
+    string(REGEX MATCH "VK_MAKE_VERSION\\(.*\\)" VulkanHeaders_out ${VulkanHeaders_line})
+    string(REGEX MATCHALL "[0-9]+" VulkanHeaders_MAJOR_MINOR "${VulkanHeaders_out}")
+    if (VulkanHeaders_MAJOR_MINOR)
+        list (GET VulkanHeaders_MAJOR_MINOR 0 VulkanHeaders_cur_major)
+        list (GET VulkanHeaders_MAJOR_MINOR 1 VulkanHeaders_cur_minor)
         if (${VulkanHeaders_cur_major} GREATER ${VulkanHeaders_VERSION_MAJOR})
             set(VulkanHeaders_VERSION_MAJOR ${VulkanHeaders_cur_major})
             set(VulkanHeaders_VERSION_MINOR ${VulkanHeaders_cur_minor})
         endif()
-        if (${VulkanHeaders_cur_major} EQUAL ${VulkanHeaders_VERSION_MAJOR} AND ${VulkanHeaders_cur_minor} GREATER ${VulkanHeaders_VERSION_MINOR})
+        if (${VulkanHeaders_cur_major} EQUAL ${VulkanHeaders_VERSION_MAJOR} AND
+            ${VulkanHeaders_cur_minor} GREATER ${VulkanHeaders_VERSION_MINOR})
             set(VulkanHeaders_VERSION_MINOR ${VulkanHeaders_cur_minor})
         endif()
     endif()
 
-    # The header version is really the patch version.
-    string(FIND "${VulkanHeaders_cur_line}" "VK_HEADER_VERSION" VulkanHeaders_found_headerversion)
-    if (${VulkanHeaders_found_headerversion} GREATER -1)
-        string(REPLACE "#define VK_HEADER_VERSION " "" VulkanHeaders_patch_line ${VulkanHeaders_cur_line})
-        string(REPLACE " " "" VulkanHeaders_VERSION_PATCH ${VulkanHeaders_patch_line})
+    # Second, handle the case where we have the patch version
+    #   Format is:
+    #      #define VK_HEADER_VERSION Z
+    #   Where Z is the patch version which we just grab off the end
+    string(REGEX MATCH "define.*VK_HEADER_VERSION.*" VulkanHeaders_out ${VulkanHeaders_line})
+    list(LENGTH VulkanHeaders_out VulkanHeaders_len)
+    if (VulkanHeaders_len)
+        string(REGEX MATCHALL "[0-9]+" VulkanHeaders_VERSION_PATCH "${VulkanHeaders_out}")
```

Comment:
This is slightly less future-proof because if they ever add a comment:

old:
```c
#define VK_HEADER_VERSION 80
```
new:
```c
#define VK_HEADER_VERSION 80  // patch/revision level 80
```

the above code will match two numbers and you'll end up with a list instead of a single string in `VulkanHeader_VERSION_PATCH`.

You can combat this in one or several of the following ways:
* Check `VulkanHeaders_len` to see if it equals 1 and abort if not
* Check `VulkanHeaders_len` to see if it is at least 1 and extract only the first element
* Change the MATCHALL TO MATCH in line 127 so that it returns only the first match
* Change the filter in 124 to strip out everything after the number.  Maybe `"define.*VK_HEADER_VERSION.*[0-9]+"`

I guess if it were me, I'd do the last two.

Reply:
Did the last two.  Also tested by adding a comment to the end just to be safe.